--- a/util/grub-setup.c	2013-05-07 19:34:19.664065508 +0800
+++ b/util/grub-setup.c	2013-05-07 19:39:25.503282411 +0800
@@ -141,7 +141,7 @@
 static void
 setup (const char *dir,
        const char *boot_file, const char *core_file,
-       const char *root, const char *dest, int force,
+       char *root, const char *dest, int force,
        int fs_probe, int allow_floppy)
 {
   char *boot_path, *core_path, *core_path_dev, *core_path_dev_full;
