--- a/arch/mips/ar71xx/machtype.h	2011-09-01 11:44:06.248089933 +0800
+++ b/arch/mips/ar71xx/machtype.h	2011-09-01 11:46:56.738088913 +0800
@@ -72,6 +72,7 @@
 	AR71XX_MACH_DB120,      /* Atheros DB120 (AR934x based) */
 	AR71XX_MACH_EZBOX_TL_WR740N_V4,	/* EZBOX on TL-WR740N v4 */
 	AR71XX_MACH_EZBOX_MW150R_V2,	/* EZBOX on Mercury MW150R v2 */
+	AR71XX_MACH_EZBOX_AP81,	/* EZBOX on Atheros AP81 */
 };
 
 #endif /* _AR71XX_MACHTYPE_H */
--- a/arch/mips/ar71xx/Kconfig	2011-09-01 11:44:22.168091038 +0800
+++ b/arch/mips/ar71xx/Kconfig	2011-09-01 11:46:05.408089798 +0800
@@ -373,6 +373,18 @@
 	select AR71XX_UBOOT_ENV
 	default n
 
+config AR71XX_MACH_EZBOX_AP81
+	bool "EZBOX on Atheros AP81 board support"
+	select SOC_AR913X
+	select AR71XX_DEV_M25P80
+	select AR71XX_DEV_AR9XXX_WMAC
+	select AR71XX_DEV_GPIO_BUTTONS
+	select AR71XX_DEV_LEDS_GPIO
+	select AR71XX_DEV_USB
+	select AR71XX_NVRAM
+	select AR71XX_UBOOT_ENV
+	default n
+
 endmenu
 
 config SOC_AR71XX
--- a/arch/mips/ar71xx/Makefile	2011-09-01 11:44:29.278090913 +0800
+++ b/arch/mips/ar71xx/Makefile	2011-09-01 11:46:27.718090508 +0800
@@ -68,3 +68,4 @@
 obj-$(CONFIG_AR71XX_MACH_ZCN_1523H)	+= mach-zcn-1523h.o
 obj-$(CONFIG_AR71XX_MACH_EZBOX_TL_WR740N_V4)	+= mach-ezbox-tl-wr740n_v4.o
 obj-$(CONFIG_AR71XX_MACH_EZBOX_MW150R_V2)	+= mach-ezbox-mw150r_v2.o
+obj-$(CONFIG_AR71XX_MACH_EZBOX_AP81)	+= mach-ezbox-ap81.o
