--- nxlib/stub.c	2010-05-21 12:11:12.650217409 +0800
+++ nxlib/stub.c	2010-05-21 12:24:50.274216647 +0800
@@ -160,3 +160,10 @@
 int XUnregisterIMInstantiateCallback() { DPRINTF("XUnregisterIMInstantiateCallback called\n"); return 0; } 
 int XIMOfIC() { DPRINTF("XIMOfIC called\n"); return 0; }
 int XLocaleOfIM() { DPRINTF("XLocaleOfIM called\n"); return 0; }
+
+/* required for fltk 2.0.x */
+int XFreeDeviceList() { printf("XFreeDeviceList called\n"); return 0; }
+int XOpenDevice() { printf("XOpenDevice called\n"); return 0; }
+int XListInputDevices() { printf("XListInputDevices called\n"); return 0; }
+int XSelectExtensionEvent() { printf("XSelectExtensionEvent called\n"); return 0; }
+int XGetExtensionVersion() { printf("XGetExtensionVersion called\n"); return 0; }
