--- nxlib/stub.c	2011-02-06 22:28:46.642717212 +0800
+++ nxlib/stub.c	2011-02-06 22:30:28.734740787 +0800
@@ -177,3 +177,7 @@
 /* required for pango 1.26.0 */
 int XExtendedMaxRequestSize() { DPRINTF("XExtendedMaxRequestSize called\n"); return 0; }
 int XInitImage() { DPRINTF("XInitImage called\n"); return 0; }
+
+/* required for gtk+ 2.17.0 */
+int XextFindDisplay() { DPRINTF("XextFindDisplay called\n"); return 0; }
+int XRenderQueryExtension() { DPRINTF("XRenderQueryExtension called\n"); return 0; }
