--- tk8.5.8.orig/unix/tcl.m4	2011-02-04 00:21:37.238734836 +0800
+++ tk8.5.8/unix/tcl.m4	2011-02-04 00:24:34.262737922 +0800
@@ -2415,16 +2415,16 @@
 	for i in $dirs ; do
 	    if test -r $i/libX11.a -o -r $i/libX11.so -o -r $i/libX11.sl -o -r $i/libX11.dylib; then
 		AC_MSG_RESULT([$i])
-		XLIBSW="-L$i -lX11"
+		XLIBSW="-L$i -lX11 -lnano-X"
 		x_libraries="$i"
 		break
 	    fi
 	done
     else
 	if test "$x_libraries" = ""; then
-	    XLIBSW=-lX11
+	    XLIBSW="-lX11 -lnano-X"
 	else
-	    XLIBSW="-L$x_libraries -lX11"
+	    XLIBSW="-L$x_libraries -lX11 -lnano-X"
 	fi
     fi
     if test "$XLIBSW" = nope ; then
@@ -2432,7 +2432,7 @@
     fi
     if test "$XLIBSW" = nope ; then
 	AC_MSG_RESULT([could not find any!  Using -lX11.])
-	XLIBSW=-lX11
+	XLIBSW="-lX11 -lnano-X"
     fi
 ])
 
