--- ape-server-1.2.0.orig/Makefile	2012-07-21 00:16:47.501689723 +0800
+++ ape-server-1.2.0/Makefile	2012-07-21 00:17:42.401687414 +0800
@@ -14,7 +14,8 @@
 all: aped
 	
 aped: $(SRC)
-	$(CC) $(CFLAGS) $(SRC) -o $(EXEC) $(LFLAGS) ./deps/udns-0.0.9/libudns.a -I ./deps/udns-0.0.9/
+	#$(CC) $(CFLAGS) $(SRC) -o $(EXEC) $(LFLAGS) ./deps/udns-0.0.9/libudns.a -I ./deps/udns-0.0.9/
+	$(CC) $(CFLAGS) $(SRC) -o $(EXEC) $(LFLAGS)
 install: 
 	install -d $(bindir)
 	install -m 755 $(EXEC) $(bindir)
