--- a/drivers/serial/Kconfig	2010-05-27 05:29:57.000000000 +0800
+++ b/drivers/serial/Kconfig	2010-10-09 11:37:38.372779954 +0800
@@ -1225,6 +1225,38 @@
 	  This driver can also be built as a module.  If so, the module
 	  will be called icom.
 
+config SERIAL_SOLOS
+	bool "Solos serial port support"
+	depends on ARCH_SOLOS
+	select SERIAL_CORE
+	help
+	  If you have a machine based on an Conexant Solos CPU you
+	  can enable its onboard serial port by enabling this option.
+
+config SERIAL_SOLOS_CONSOLE
+	bool "Console on Solos serial port"
+	depends on SERIAL_SOLOS
+	select SERIAL_CORE_CONSOLE
+	help
+	  If you have enabled the serial port on the Solos
+	  CPU you can make it the console by answering Y to this option.
+
+	  Even if you say Y here, the currently visible virtual console
+	  (/dev/tty0) will still be used as the system console by default, but
+	  you can alter that using a kernel command line option such as
+	  "console=ttyS0". (Try "man bootparam" or see the documentation of
+	  your boot loader (lilo or loadlin) about how to pass options to the
+	  kernel at boot time.)
+
+config SERIAL_SOLOS_CONSOLE_BAUD
+	int "Conexant Solos serial port baud"
+	depends on SERIAL_SOLOS_CONSOLE=y
+	default "38400"
+	help
+	  Select the Solos console baud rate. The default is chosen to match 
+	  what the UART will be set to at power on.
+		  
+
 config SERIAL_M32R_SIO
 	bool "M32R SIO I/F"
 	depends on M32R
--- a/drivers/serial/Makefile	2010-05-27 05:29:57.000000000 +0800
+++ b/drivers/serial/Makefile	2010-10-09 11:37:38.372779954 +0800
@@ -60,6 +60,7 @@
 obj-$(CONFIG_SERIAL_CPM) += cpm_uart/
 obj-$(CONFIG_SERIAL_IMX) += imx.o
 obj-$(CONFIG_SERIAL_MPC52xx) += mpc52xx_uart.o
+obj-$(CONFIG_SERIAL_SOLOS) += solos_serial.o
 obj-$(CONFIG_SERIAL_ICOM) += icom.o
 obj-$(CONFIG_SERIAL_M32R_SIO) += m32r_sio.o
 obj-$(CONFIG_SERIAL_MPSC) += mpsc.o
