--- freeswitch-1.0.6.orig/Makefile.am	2010-06-30 11:28:29.699078599 +0800
+++ freeswitch-1.0.6/Makefile.am	2010-06-30 11:29:36.855078419 +0800
@@ -253,7 +253,7 @@
 $(libfreeswitch_la_SOURCES): $(CORE_LIBS) $(switch_builddir)/quiet_libtool $(switch_builddir)/modules.conf  
 
 src/include/switch_swigable_cpp.h: $(switch_srcdir)/src/include/switch_cpp.h
-	$(CC) -E $(switch_srcdir)/src/include/switch_cpp.h -DSWITCH_DECLARE_CLASS= -DSWITCH_DECLARE\(x\)=x -DSWITCH_DECLARE_CONSTRUCTOR= -DSWITCH_DECLARE_NONSTD\(x\)=x 2>/dev/null | grep -v "^#" > src/include/switch_swigable_cpp.h 
+	$(CC) -E $(switch_srcdir)/src/include/switch_cpp.h -I$(switch_srcdir)/src/include -I$(switch_srcdir)/libs/libteletone/src -DSWITCH_DECLARE_CLASS= -DSWITCH_DECLARE\(x\)=x -DSWITCH_DECLARE_CONSTRUCTOR= -DSWITCH_DECLARE_NONSTD\(x\)=x 2>/dev/null | grep -v "^#" > src/include/switch_swigable_cpp.h 
 
 
 ##
--- freeswitch-1.0.6.orig/Makefile.in	2010-06-30 11:29:44.127077163 +0800
+++ freeswitch-1.0.6/Makefile.in	2010-06-30 11:30:07.283075650 +0800
@@ -1995,7 +1995,7 @@
 $(libfreeswitch_la_SOURCES): $(CORE_LIBS) $(switch_builddir)/quiet_libtool $(switch_builddir)/modules.conf  
 
 src/include/switch_swigable_cpp.h: $(switch_srcdir)/src/include/switch_cpp.h
-	$(CC) -E $(switch_srcdir)/src/include/switch_cpp.h -DSWITCH_DECLARE_CLASS= -DSWITCH_DECLARE\(x\)=x -DSWITCH_DECLARE_CONSTRUCTOR= -DSWITCH_DECLARE_NONSTD\(x\)=x 2>/dev/null | grep -v "^#" > src/include/switch_swigable_cpp.h 
+	$(CC) -E $(switch_srcdir)/src/include/switch_cpp.h -I$(switch_srcdir)/src/include -I$(switch_srcdir)/libs/libteletone/src -DSWITCH_DECLARE_CLASS= -DSWITCH_DECLARE\(x\)=x -DSWITCH_DECLARE_CONSTRUCTOR= -DSWITCH_DECLARE_NONSTD\(x\)=x 2>/dev/null | grep -v "^#" > src/include/switch_swigable_cpp.h 
 
 scripts/fsxs: scripts/fsxs.in
 	@echo creating fsxs
