--- libvirt-1.0.2.orig/Makefile.am	2013-02-21 23:18:07.902476382 +0800
+++ libvirt-1.0.2/Makefile.am	2013-02-21 23:19:27.782476694 +0800
@@ -7,9 +7,7 @@
 GENHTML = genhtml
 
 SUBDIRS = gnulib/lib include src daemon tools docs gnulib/tests \
-  python tests po examples/domain-events/events-c examples/hellolibvirt \
-  examples/dominfo examples/domsuspend examples/python examples/apparmor \
-  examples/xml/nwfilter examples/openauth examples/systemtap
+  python tests
 
 ACLOCAL_AMFLAGS = -I m4 -I gnulib/m4
 
@@ -26,10 +24,8 @@
   Makefile.nonreentrant \
   autogen.sh \
   cfg.mk \
-  examples/domain-events/events-python \
   run.in \
-  AUTHORS.in \
-  $(XML_EXAMPLES)
+  AUTHORS.in
 
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = libvirt.pc
