--- Mesa-7.9.2.orig/configure.ac	2011-09-30 23:05:05.368250937 +0800
+++ Mesa-7.9.2/configure.ac	2011-09-30 23:08:31.918253750 +0800
@@ -605,8 +605,8 @@
         GL_PC_LIB_PRIV="$GL_LIB_DEPS"
         GL_PC_CFLAGS="$X11_INCLUDES"
     fi
-    GL_LIB_DEPS="$GL_LIB_DEPS $SELINUX_LIBS -lm -lpthread"
-    GL_PC_LIB_PRIV="$GL_PC_LIB_PRIV $SELINUX_LIBS -lm -lpthread"
+    GL_LIB_DEPS="$GL_LIB_DEPS $SELINUX_LIBS -lm -lmisc -lpthread"
+    GL_PC_LIB_PRIV="$GL_PC_LIB_PRIV $SELINUX_LIBS -lm -lmisc -lpthread"
 
     # if static, move the external libraries to the programs
     # and empty the libraries for libGL
@@ -664,12 +664,12 @@
     fi
 
     # need DRM libs, -lpthread, etc.
-    GL_LIB_DEPS="$GL_LIB_DEPS $LIBDRM_LIBS -lm -lpthread $DLOPEN_LIBS"
-    GL_PC_LIB_PRIV="-lm -lpthread $DLOPEN_LIBS"
-    GLESv1_CM_LIB_DEPS="$LIBDRM_LIBS -lm -lpthread $DLOPEN_LIBS"
-    GLESv1_CM_PC_LIB_PRIV="-lm -lpthread $DLOPEN_LIBS"
-    GLESv2_LIB_DEPS="$LIBDRM_LIBS -lm -lpthread $DLOPEN_LIBS"
-    GLESv2_PC_LIB_PRIV="-lm -lpthread $DLOPEN_LIBS"
+    GL_LIB_DEPS="$GL_LIB_DEPS $LIBDRM_LIBS -lm -lmisc -lpthread $DLOPEN_LIBS"
+    GL_PC_LIB_PRIV="-lm -lmisc -lpthread $DLOPEN_LIBS"
+    GLESv1_CM_LIB_DEPS="$LIBDRM_LIBS -lm -lmisc -lpthread $DLOPEN_LIBS"
+    GLESv1_CM_PC_LIB_PRIV="-lm -lmisc -lpthread $DLOPEN_LIBS"
+    GLESv2_LIB_DEPS="$LIBDRM_LIBS -lm -lmisc -lpthread $DLOPEN_LIBS"
+    GLESv2_PC_LIB_PRIV="-lm -lmisc -lpthread $DLOPEN_LIBS"
     ;;
 osmesa)
     # No libGL for osmesa
@@ -898,7 +898,7 @@
         [AC_MSG_ERROR([Expat required for DRI.])])
 
     # put all the necessary libs together
-    DRI_LIB_DEPS="$SELINUX_LIBS $LIBDRM_LIBS $EXPAT_LIB -lm -lpthread $DLOPEN_LIBS"
+    DRI_LIB_DEPS="$SELINUX_LIBS $LIBDRM_LIBS $EXPAT_LIB -lm -lmisc -lpthread $DLOPEN_LIBS"
 fi
 AC_SUBST([DRI_DIRS])
 AC_SUBST([EXPAT_INCLUDES])
@@ -971,12 +971,12 @@
 *osmesa*)
     # only link libraries with osmesa if shared
     if test "$enable_static" = no; then
-        OSMESA_LIB_DEPS="-lm -lpthread $SELINUX_LIBS $DLOPEN_LIBS"
+        OSMESA_LIB_DEPS="-lm -lmisc -lpthread $SELINUX_LIBS $DLOPEN_LIBS"
     else
         OSMESA_LIB_DEPS=""
     fi
     OSMESA_MESA_DEPS=""
-    OSMESA_PC_LIB_PRIV="-lm -lpthread $SELINUX_LIBS $DLOPEN_LIBS"
+    OSMESA_PC_LIB_PRIV="-lm -lmisc -lpthread $SELINUX_LIBS $DLOPEN_LIBS"
     ;;
 esac
 AC_SUBST([OSMESA_LIB_DEPS])
@@ -1049,9 +1049,9 @@
     *)
         # If static, empty GLU_LIB_DEPS and add libs for programs to link
         GLU_PC_REQ="gl"
-        GLU_PC_LIB_PRIV="-lm"
+        GLU_PC_LIB_PRIV="-lm -lmisc"
         if test "$enable_static" = no; then
-            GLU_LIB_DEPS="-lm"
+            GLU_LIB_DEPS="-lm -lmisc"
             GLU_MESA_DEPS='-l$(GL_LIB)'
         else
             GLU_LIB_DEPS=""
@@ -1183,8 +1183,8 @@
     if test "x$GCC" = xyes; then
         GLUT_CFLAGS="$GLUT_CFLAGS -fexceptions"
     fi
-    GLUT_LIB_DEPS="$GLUT_LIB_DEPS -lm"
-    GLUT_PC_LIB_PRIV="$GLUT_PC_LIB_PRIV -lm"
+    GLUT_LIB_DEPS="$GLUT_LIB_DEPS -lm -lmisc"
+    GLUT_PC_LIB_PRIV="$GLUT_PC_LIB_PRIV -lm -lmisc"
 
     # If static, empty GLUT_LIB_DEPS and add libs for programs to link
     if test "$enable_static" = no; then
@@ -1210,13 +1210,13 @@
 if test "x$APP_LIB_DEPS" = x; then
     case "$host_os" in
     solaris*)
-        APP_LIB_DEPS="-lX11 -lsocket -lnsl -lm"
+        APP_LIB_DEPS="-lX11 -lsocket -lnsl -lm -lmisc"
         ;;
     cygwin*)
         APP_LIB_DEPS="-lX11"
         ;;
     *)
-        APP_LIB_DEPS="-lm"
+        APP_LIB_DEPS="-lm -lmisc"
         ;;
     esac
 fi
--- Mesa-7.9.2.orig/src/glsl/Makefile	2011-09-30 23:42:25.807652894 +0800
+++ Mesa-7.9.2/src/glsl/Makefile	2011-09-30 23:43:16.500491338 +0800
@@ -87,6 +87,8 @@
 LIBS = \
 	$(TOP)/src/glsl/libglsl.a
 
+LIBS += -lm -lmisc
+
 APPS = glsl_compiler glcpp/glcpp
 
 GLSL2_C_SOURCES = \
