--- nxlib/stub.c	2011-06-06 21:41:28.072154687 +0800
+++ nxlib/stub.c	2011-06-06 21:42:25.063172842 +0800
@@ -194,3 +194,6 @@
 int XRenderFindFormat() { DPRINTF("XRenderFindFormat called\n"); return 0; }
 int _XGetAsyncReply() { DPRINTF("_XGetAsyncReply called\n"); return 0; }
 int XShapeQueryVersion() { DPRINTF("XShapeQueryVersion called\n"); return 0; }
+
+/* required for libXft */
+int XRenderFindStandardFormat() { DPRINTF("XRenderFindStandardFormat called\n"); return 0; }
