--- linux-2.6.32.14/arch/arm/Kconfig	2010-05-27 05:29:57.000000000 +0800
+++ linux-2.6.32.14-solos/arch/arm/Kconfig	2010-10-01 23:38:15.778480343 +0800
@@ -304,6 +307,14 @@
 	  Support for systems based on the DC21285 companion chip
 	  ("FootBridge"), such as the Simtec CATS and the Rebel NetWinder.
 
+config ARCH_SOLOS
+	bool "Conexant Solos"
+	select CPU_ARM926T
+	select USB_ARCH_HAS_EHCI
+	help
+	  Select this if you want support for Conesant Solos family of SoCs. Otherwise
+	  say 'N'
+
 config ARCH_MXC
 	bool "Freescale MXC/iMX-based"
 	select GENERIC_TIME
@@ -776,6 +787,8 @@
 
 source "arch/arm/mach-h720x/Kconfig"
 
+source "arch/arm/mach-solos/Kconfig"
+
 source "arch/arm/mach-versatile/Kconfig"
 
 source "arch/arm/mach-aaec2000/Kconfig"
--- linux-2.6.32.14/arch/arm/Makefile	2010-05-27 05:29:57.000000000 +0800
+++ linux-2.6.32.14-solos/arch/arm/Makefile	2010-10-01 23:39:30.558483838 +0800
@@ -126,6 +126,13 @@
 machine-$(CONFIG_ARCH_EP93XX)		:= ep93xx
 machine-$(CONFIG_ARCH_GEMINI)		:= gemini
 machine-$(CONFIG_ARCH_H720X)		:= h720x
+machine-$(CONFIG_ARCH_SOLOS)		:= solos
+ifdef CONFIG_ARCH_SOLOSW_GALLUS
+textaddr-$(CONFIG_ARCH_SOLOS)		:= 0xc0028000
+endif
+ifdef CONFIG_ARCH_SOLOS_GALLUS
+textaddr-$(CONFIG_ARCH_SOLOS)		:= 0xc0018000
+endif
 machine-$(CONFIG_ARCH_INTEGRATOR)	:= integrator
 machine-$(CONFIG_ARCH_IOP13XX)		:= iop13xx
 machine-$(CONFIG_ARCH_IOP32X)		:= iop32x
