--- a/arch/arm/include/asm/cacheflush.h	2010-05-27 05:29:57.000000000 +0800
+++ b/arch/arm/include/asm/cacheflush.h	2010-09-29 09:09:08.000000000 +0800
@@ -42,7 +43,8 @@
 #endif
 
 #if defined(CONFIG_CPU_ARM920T) || defined(CONFIG_CPU_ARM922T) || \
-    defined(CONFIG_CPU_ARM925T) || defined(CONFIG_CPU_ARM1020)
+    defined(CONFIG_CPU_ARM925T) || defined(CONFIG_CPU_ARM1020) || \
+    defined(CONFIG_CPU_ARM1026)
 # define MULTI_CACHE 1
 #endif
 
