--- a/configure	2011-10-15 12:10:35.171070325 +0800
+++ b/configure	2011-10-15 12:17:01.990425059 +0800
@@ -235,6 +235,9 @@
   if test "`($CC -w -c $SFLAGS $test.c) 2>&1`" = "" &&
      test "`($LDSHARED $SFLAGS -o $test$shared_ext $test.o) 2>&1`" = ""; then
     echo Building shared library $SHAREDLIBV with $CC.
+  elif test "`($CC -w -c $test.c) 2>&1`" = "" &&
+     test "`($LDSHARED -o $test$shared_ext $test.o) 2>&1`" = ""; then
+    echo Building shared library $SHAREDLIBV with $CC.
   elif test -z "$old_cc" -a -z "$old_cflags"; then
     echo No shared library support.
     shared=0;
