--- redis-2.4.15.orig/src/config.h	2012-07-12 14:49:14.239132604 +0800
+++ redis-2.4.15/src/config.h	2012-07-12 14:52:47.859141460 +0800
@@ -25,9 +25,13 @@
 #endif
 
 /* Test for backtrace() */
+#if defined(__UCLIBC__)
+#undef HAVE_BACKTRACE
+#else
 #if defined(__APPLE__) || defined(__linux__) || defined(__sun)
 #define HAVE_BACKTRACE 1
 #endif
+#endif
 
 /* Test for polling API */
 #ifdef __linux__
