--- tk8.5.8/unix/Makefile.in	2011-06-08 12:57:30.527654002 +0800
+++ tk8.5.8/unix/Makefile.in	2011-06-08 12:59:14.059653622 +0800
@@ -52,7 +52,7 @@
 TK_LIBRARY		= @TK_LIBRARY@
 
 # Path to use at runtime to refer to LIB_INSTALL_DIR:
-LIB_RUNTIME_DIR		= $(libdir)
+LIB_RUNTIME_DIR		= $(TARGET_LIB_RUNTIME_DIR)
 
 # Directory in which to install the program wish:
 BIN_INSTALL_DIR		= $(INSTALL_ROOT)$(bindir)
