--- microwin/src/Arch.rules	2010-06-04 13:05:13.580919102 +0800
+++ microwin/src/Arch.rules	2010-06-04 13:05:37.448918827 +0800
@@ -23,7 +23,7 @@
 CXX_COMPILER = g++
 TOOLSPREFIX = $(NATIVETOOLSPREFIX)
 DEFINES += -DLINUX=1 -DUNIX=1
-CFLAGS += $(GCC_WARNINGS) $(OPTFLAGS)
+CFLAGS += $(GCC_WARNINGS) $(OPTFLAGS) -D_GNU_SOURCE
 LDFLAGS +=
 # include the following line for static libraries
 #LDFLAGS += -static
