--- a/drivers/usb/host/Kconfig	2010-10-09 23:25:24.413475399 +0800
+++ b/drivers/usb/host/Kconfig	2010-10-09 23:29:12.125479241 +0800
@@ -59,6 +59,22 @@
 	  To compile this driver as a module, choose M here: the
 	  module will be called ehci-hcd.
 
+config SOLOS_USB_HOST0
+	bool "Enable USB host 0"
+	depends on USB && ARCH_SOLOS
+
+config SOLOS_USB_HOST1
+	bool "Enable USB host 1"
+	depends on USB && ARCH_SOLOS && !ARCH_SOLOS_228PIN
+
+config USB_SOLOS_PHY0_EXT_CLK
+	bool "Use external clock on PHY 0"
+	depends on USB && ARCH_SOLOS && SOLOS_USB_HOST0
+
+config USB_SOLOS_PHY1_EXT_CLK
+	bool "Use external clock on PHY 1"
+	depends on USB && ARCH_SOLOS && SOLOS_USB_HOST1
+
 config USB_EHCI_ROOT_HUB_TT
 	bool "Root Hub Transaction Translators"
 	depends on USB_EHCI_HCD
--- a/drivers/usb/host/ehci-hcd.c	2010-10-11 09:06:49.256137561 +0800
+++ b/drivers/usb/host/ehci-hcd.c	2010-10-11 09:08:28.522766134 +0800
@@ -1142,6 +1142,11 @@
 #define	PLATFORM_DRIVER		ehci_atmel_driver
 #endif
 
+#ifdef CONFIG_ARCH_SOLOS
+#include "ehci-solos.c"
+#define	PLATFORM_DRIVER		ehci_hcd_solos_driver
+#endif
+
 #if !defined(PCI_DRIVER) && !defined(PLATFORM_DRIVER) && \
     !defined(PS3_SYSTEM_BUS_DRIVER) && !defined(OF_PLATFORM_DRIVER)
 #error "missing bus glue for ehci-hcd"
